spec/filters/elasticsearch_spec.rb

```diff
@@ -82,6 +82,24 @@
         expect {plugin.register}.to raise_error(LogStash::ConfigurationError)
       end
     end
+
+
+    context "with custom headers" do
+      let(:config) do
+        {
+          "hosts" => ["localhost:9200"],
+          "query" => "response: 404",
+          "custom_headers" => { "Custom-Header-1" => "Custom Value 1", "Custom-Header-2" => "Custom Value 2" }
+        }
+      end
+
+
+      it "sets custom headers" do
+        plugin.register
+        client = plugin.send(:client)
```

Comment:
We are almost here (sorry, I missed this).
```suggestion
        client = plugin.send(:get_client).client
```